File: sast-python2/case/completeness/single_app_tracing/datatype/map/map_001_T.py
# -*- coding: utf-8 -*-
# evaluation information start
# real case = true
# evaluation item = 完整度->单应用跟踪完整度->数据类型和结构->字典
# scene introduction = 字典/映射（Map）对象
# level = 2
# bind_url = completeness/single_app_tracing/datatype/map/map_001_T
# evaluation information end
import os


def map_001_T(taint_src):
    m = dict()  # 或者直接使用 m = {}
    m[u'key1'] = taint_src
    taint_sink(m)


def taint_sink(o):
    os.system(unicode(o))


if __name__ == u"__main__":
    taint_src = u"taint_src_value"
    map_001_T(taint_src)

File: sast-js/case/completeness/single_app_tracing/control_flow/loop_stmt/map_001_T.js
// evaluation information start
// real case = true
// evaluation item = 完整度->单应用跟踪完整度->流程控制语句->循环结构
// scene introduction = map循环
// level = 2
// bind_url = completeness/single_app_tracing/control_flow/loop_stmt/map_001_T
// date = 2025-12-18 06:02:26
// evaluation information end
const { execSync } = require("child_process");

function map_001_T(__taint_src) {
  let arr = ["safe_value", __taint_src];
  // 场景特点：使用map方法遍历数组并传播污点数据
  let result = arr.map((item) => item);
  __taint_sink(result);
}

function __taint_sink(o) {
  execSync(JSON.stringify(o));
}

const taint_src = "taint_src_value";
map_001_T(taint_src);
